### ISSUE TITLE
Chart fill parent dimension

### ISSUE THREAD

**Issue by besil** [indiespirit/react-native-chart-kit#146]
Is it possible to make a chart fit the parent component dimension?
Without specifying the width or height attribute, the chart crashes (NaN provided).

I'd like to wrap the chart in a Card component, with custom borders and so on, but I don't understand how to correctly wrap it.

Thank you for this great library, it's one of my favourite!
S.

**Comment by athenawisdoms** [comment]
@besil  Did you manage to get the chart to fill up 100% of the parent container's dimensions? 

Tried the following but the app throws an error about `Invariant Violation`

```
<LineChart
    height={'100%'}
/>
```

**Comment by besil** [comment]
Hi, I just used 
`const screenWidth = Dimensions.get('window').width;`

I'd have preferred  a 100% option, but it doesn't seem to work

**Comment by rishab1065** [comment]
Any updates on this issue?

**Comment by vedmant** [comment]
Yeah, I have an issue with this too, I need to manually remove padding from all parent containers to make chart not overflow parents.

**Comment by rvkuzmik** [comment]
Still an issue in 2021.. No one seems to know how to get this working.. My other option might be to somehow get the dimensions of parent View and go from there..

**Comment by gpessa** [comment]
Any update?

**Comment by rvkuzmik** [comment]
> Any update?

I did it somehow, don't remember how though. When I get back to work tomorrow I can check it out and update this thread.

**Comment by gpessa** [comment]
You are my hero!

**Comment by YanvegHD** [comment]
> > Any update?
> 
> I did it somehow, don't remember how though. When I get back to work tomorrow I can check it out and update this thread.

Are you still at work ? 😆

**Comment by Boscotec** [comment]
See solution by mathcale https://github.com/indiespirit/react-native-chart-kit/issues/506#issuecomment-917564118

**Comment by MadelinFg** [comment]
> See solution by mathcale [#506 (comment)](https://github.com/indiespirit/react-native-chart-kit/issues/506#issuecomment-917564118)

any other solution so far? I've been facing this issue for days 😞
